src/palace/manager/sqlalchemy/util.py

```diff
@@ -138,7 +140,16 @@ def get_one_or_create(
                 kwargs,
                 e,
             )
-            return db.query(model).filter_by(**kwargs).one(), False
+            one = get_one(db, model, **kwargs)
+            if one is None:
+                # If we still don't have an object, raise the error.
+                log.error(
+                    "Unable to retrieve with get_one after IntegrityError with %r %r",
+                    model,
+                    kwargs,
+                )
+                raise
+            return one, False
```

Comment:
These changes are not directly related, but they fix a bug in the existing code that the changes in this PR surfaced. In the case where you are passing `constraint` to `get_one_or_create`, if there is an `IntegrityError` previously we would ignore the `constraint` when doing the second query. 

We likely wouldn't hit this often / ever in the real world, but good to handle that case, since i noticed it as part of these changes.